src/main/java/oscar/oscarLab/ca/all/util/Utilities.java

```diff
@@ -113,38 +117,32 @@ public static ArrayList<String> separateMessages(String fileName) throws Excepti
      * @param filename
      * @return String
      */
-    public static String saveFile(InputStream stream,String filename ){
+    public static String saveFile(InputStream stream, String filename) {
         String retVal = null;
         
-        
         try {
             OscarProperties props = OscarProperties.getInstance();
-            //properties must exist
-            String place= props.getProperty("DOCUMENT_DIR");
-            
-            if(!place.endsWith("/"))
-                place = new StringBuilder(place).insert(place.length(),"/").toString();
-            retVal = place+"LabUpload."+filename.replaceAll(".enc", "")+"."+(new Date()).getTime();
-            
-            logger.debug("saveFile place="+place+", retVal="+retVal);
-            //write the  file to the file specified
-            OutputStream os = new FileOutputStream(retVal);
-            
-            int bytesRead = 0;
-            while ((bytesRead = stream.read()) != -1){
-                os.write(bytesRead);
+            String place = props.getProperty("DOCUMENT_DIR");
+
+            if (!place.endsWith("/")) { place += "/"; }
```

Comment:
Please do better than the last developer and use the Paths utility to build the URI. ie: `retVal = Paths.get(place, "LabUpload." + filename.replaceAll(".enc", "") + "." + System.currentTimeMillis()")`
Doing this will guarantee a safe filepath and eliminate the need for this mess:
`if (!place.endsWith("/")) { place += "/"; }`